scipy/stats/_continuous_distns.py

```diff
@@ -6941,6 +6941,161 @@ def _stats(self, a):
 
     def _entropy(self, a):
         return 1 - 1.0/a - np.log(a)
+    
+    def _support_mask(self, x, a):
+        if np.any(a < 1):
+            return (x != 0) & super(powerlaw_gen, self)._support_mask(x, a)
+        else:
+            return super(powerlaw_gen, self)._support_mask(x, a)
+
+    def fit(self, data, *args, **kwds):
+        '''
+        The strategy for this fit method is complex, so its steps will be
+        described here first.
```

Comment:
Please add the steps corresponding with some parameters being fixed. For instance, if scale and location are both fixed, you return the shape before doing any of the things described here.